src/command_transfer.h

```diff
@@ -0,0 +1,63 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2015 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include "usb/usb-device.h"
+
+#include <vector>
+#include <stdint.h>
+
+namespace librealsense
+{
+    namespace platform
+    {
+        class command_transfer
+        {
+        public:
+            virtual std::vector<uint8_t> send_receive(
+                const std::vector<uint8_t>& data,
+                int timeout_ms = 5000,
+                bool require_response = true) = 0;
+
+            virtual ~command_transfer() = default;
+        };
+
+        class command_transfer_usb : public command_transfer
+        {
+        public:
+            command_transfer_usb(rs_usb_device device) : _device(device) {}
+            ~command_transfer_usb() {};
+
+            std::vector<uint8_t> send_receive(
+                const std::vector<uint8_t>& data,
+                int timeout_ms = 5000,
```

Comment:
Pls refactor this into class member with capability for overwrite via ctor param or otherwise (use-case of full FW dump/burn)

Reply:
let's talk